Scripts/datatypes/purpose.py

```diff
@@ -583,3 +583,26 @@ def calc_traffic(self, impedance: dict, purpose_key: str):
         generation = numpy.tile(self.zone_data[f"gen_{purpose_key}"], (nr_zones, 1))
         demand = {mode: (probs.pop(mode) * generation).T for mode in self.modes}
         return demand
+
+    def calc_vehicles(self, matrix: numpy.ndarray, ass_class: str, args: dict):
+        """Calculate vehicle matrix from ton matrix using ton-to-vehicles 
+        conversion values.
+
+        Parameters
+        ----------
+        matrix : numpy.ndarray
+            ton matrix
+        ass_class : str
+            truck assignment class
+        args : dict
+            conversion variable : float
+
+        Returns
+        -------
+        numpy.ndarray
+            assignment class specific vehicle matrix
+        """
+        vehicles = (matrix * args[f"{ass_class}_share"] 
+                    / args[f"{ass_class}_avg_load"] * 1.02 / 365)
```

Comment:
What does 1.02 represent?

Reply:
This is an addon factor which has been explained to me to signify uncertainty in the function. I wasn't really content with this approach either because it's hard coded. I think I'm going to first raise a question to the others on the necessity of this one.